libs/ktxreader/include/ktxreader/Ktx2Reader.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef KTXREADER_KTX2READER_H
+#define KTXREADER_KTX2READER_H
+
+#include <cstdint>
+#include <cstddef>
+
+#include <filament/Texture.h>
+
+#include <utils/FixedCapacityVector.h>
+
+namespace filament {
+    class Engine;
+}
+
+namespace basist {
+    class ktx2_transcoder;
+}
+
+namespace ktxreader {
+
+class Ktx2Reader {
+    public:
+        using Engine = filament::Engine;
+        using Texture = filament::Texture;
+        enum TransferFunction { LINEAR, sRGB };
+
+        Ktx2Reader(Engine& engine, bool quiet = false);
+        ~Ktx2Reader();
+
+        /**
+         * Requests that the reader constructs Filament textures with given internal format.
+         *
+         * This MUST be called at least once before calling load().
```

Comment:
Could we instead specify a default format?

Reply:
There might be cases where the caller considers the use of a certain formats to be "wrong" so I feel we shouldn't make assumptions.  We could maybe make it less tedious for simple cases, by allowing a format to be specified in the constructor.  The problem there is that I'm not sure what its priority should be.  Since formats are requested in priority order, would the default format be highest priority or lowest priority?  Probably the latter, but I don't think the API would be as obvious and self-explanatory as it is now.

Thanks for the great feedback in this PR by the way!  Working on fixing all the other stuff you noticed.

Reply:
Got it, makes sense! 👍